File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/PEFileInfo.kt
package com.jetbrains.signatureverifier.serialization

import com.jetbrains.signatureverifier.PeFile
import com.jetbrains.signatureverifier.crypt.SignedMessage
import kotlinx.serialization.Serializable

@Serializable
class PEFileInfo : FileInfo {
  override val signedDataInfo: SignedDataInfo
  override val fileMetaInfo: FileMetaInfo

  constructor(peFile: PeFile) {
    fileMetaInfo = peFile.getSignatureMetainfo()

    val signatureData = peFile.GetSignatureData()
    val signedMessage = SignedMessage.CreateInstance(signatureData)

    val signedData = signedMessage.SignedData

    signedDataInfo = SignedDataInfo(signedData)
  }
}

File: net/JetBrains.Serialization/src/FileInfos/PE/PeFileInfo.cs
using JetBrains.FormatRipper.Pe;
using JetBrains.SignatureVerifier.Crypt;
using Newtonsoft.Json;

namespace JetBrains.Serialization.FileInfos.PE;

[JsonObject(MemberSerialization.OptIn)]
public class PeFileInfo : FileInfo
{
  [JsonProperty("fileMetaInfo")] public override IFileMetaInfo FileMetaInfo { get; }

  [JsonProperty("signedDataInfo")] public override SignedDataInfo SignedDataInfo { get; }

  [JsonConstructor]
  public PeFileInfo(IFileMetaInfo fileMetaInfo, SignedDataInfo signedDataInfo)
  {
    FileMetaInfo = fileMetaInfo;
    SignedDataInfo = signedDataInfo;
  }

  public PeFileInfo(PeFile peFile)
  {
    FileMetaInfo = new PeFileMetaInfo(peFile.FileMetadata ??
                                      throw new ArgumentNullException("peFile.FileMetadata",
                                        "FileMetadata can not be null"));

    SignedMessage signedMessage = SignedMessage.CreateInstance(peFile.SignatureData);

    var signedData = signedMessage.SignedData;

    SignedDataInfo = new SignedDataInfo(signedData);
  }
}